### ISSUE TITLE
Reproduceable `trial.suggest_float` with `TPESampler`

### ISSUE THREAD

**Issue by kleingeo** [optuna/optuna#6344]
### Don't use GitHub Issues to ask support questions.

I am trying to optimize a model with many different hyperparameter combinations. I would also like the experiments to be reproduceable. However, when using the `optuna.samplers.TPESampler` I am getting different results with the `trial.suggest_float` when re-running the code. Here is a snippet of it.

```python
import sklearn
import numpy as np
import random
import optuna
import functools



def objective(trial, seed)

    n_estimators = trial.suggest_int("n_estimators", 10, 1000, step=10)
    criterion = trial.suggest_categorical("criterion", ["gini", "entropy", "log_loss"])
    
    
    clf = sklearn.ensemble.RandomForestClassifier(n_estimators=n_estimators,
                                                  criterion=criterion
                                                  random_state=seed
                                                  )
    
    breast_data = sklearn.datasets.load_breast_cancer()

    X = breast_data.data
    y = breast_data.target
    score = sklearn.model_selection.cross_val_score(clf, X, y, cv=10, scoring='roc_auc').mean()
    
    return score
    
    
if __name__ == '__main__':
    
    seed_int = 0
    
    random.seed(seed_int)
    np.random.seed(seed_int)
    np_rand_state = np.random.RandomState(seed_int)
    
    sampler = optuna.samplers.TPESampler(self.seed_int)
    
    study = optuna.create_study(sampler=self.sampler,
                                         direction="maximize",
                                         study_name=f"test_study",
                                         )

    study.set_metric_names(["auc"])
    
    
    partial_obj = functools.partial(objective, seed=np_rand_state)
    
    study.optimize(partial_obj, n_trials=100, n_workers=1)

````

When re-running this in debug mode, the `n_estimators ` is different each time, even when using the correct seeding and sampler. 

Is this the wrong approach to this, or is this expected behaviour given the setup?

**Comment by kleingeo** [comment]
I was able to solve this, so I'm posting the solution.

The first argument of `optuna.samplers.TPESampler` is not `seed` and it needs to be properly specified. 

```python

...

 sampler = optuna.samplers.TPESampler(seed=self.seed_int)

...
```

Simple solution: properly specify the `seed` argument.
